prometheus/src/main/scala/filodb/prometheus/query/PrometheusModel.scala

```diff
@@ -67,7 +67,9 @@ object PrometheusModel {
       b.addLabels(LabelPair.newBuilder().setName(lv._1.toString).setValue(lv._2.toString))
     }
     srv.rows.foreach { row =>
-      b.addSamples(Sample.newBuilder().setTimestampMs(row.getLong(0)).setValue(row.getDouble(1)))
+      if (!row.getDouble(1).isNaN) {
```

Comment:
Won't this prevent NaNs from being reported for raw sample queries?

Reply:
Good catch. I intended this only for HTTP API, not for raw time series. Applying this fix only for HTTP results.

Ideally, I would like to not apply this for HTTP results as well, but until we are able to set "NA" as a value in our vectors, we may have to continue to use NaN and filter them from results.